pkg/git/libgit2/checkout.go

```diff
@@ -155,21 +145,51 @@ func (c *CheckoutBranch) Checkout(ctx context.Context, path, url string, opts *g
 }
 
 type CheckoutTag struct {
-	Tag string
+	Tag          string
+	LastRevision string
 }
 
 func (c *CheckoutTag) Checkout(ctx context.Context, path, url string, opts *git.AuthOptions) (*git.Commit, error) {
-	repo, err := git2go.Clone(url, path, &git2go.CloneOptions{
-		FetchOptions: git2go.FetchOptions{
-			DownloadTags:    git2go.DownloadTagsAll,
+	repo, remote, err := getBlankRepoAndRemote(ctx, path, url, opts)
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer repo.Free()
+	defer remote.Free()
+	defer remote.Disconnect()
+
+	if c.LastRevision != "" {
+		heads, err := remote.Ls(c.Tag)
+		if err != nil {
+			return nil, fmt.Errorf("unable to remote ls for '%s': %w", managed.EffectiveURL(url), gitutil.LibGit2Error(err))
+		}
+		if len(heads) > 0 {
+			currentRevision := fmt.Sprintf("%s/%s", c.Tag, heads[0].Id.String())
+			currentAnnotatedRevision := fmt.Sprintf("%s/%s", c.Tag, heads[1].Id.String())
```

Comment:
May want to change the if for `len(heads) > 1`, as we are touching index 1.
```suggestion
			currentAnnotatedRevision := fmt.Sprintf("%s/%s", c.Tag, heads[1].Id.String())
```